services/api/database/migrations/20240115000000_taskdeployment_source.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = async function(knex) {
+    return knex.schema
+    .alterTable('task', function (table) {
+        table.enu('source_type', ['api']);
```

Comment:
acknowledging that there may be future different ways to trigger a task, to be handled in future ENUM migrations